internal/logs/log_container.go

```diff
@@ -0,0 +1,94 @@
+package logs
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+)
+
+const (
+	elapsedTime = 100 * time.Millisecond
+)
+
+type LogEntry struct {
+	Time     time.Time
+	data     []byte
+	workload string
+}
+
+func NewLogEntry(data []byte, workload string) *LogEntry {
+	return &LogEntry{
+		data:     data,
+		Time:     time.Now(),
+		workload: workload,
+	}
+}
+
+func (l *LogEntry) Size() int {
+	return len(l.data)
+}
+
+func (l *LogEntry) String() string {
+	if l.data == nil {
+		return ""
+	}
+	return string(l.data)
+}
+
+func (l *LogEntry) GetWorkload() string {
+	return l.workload
+}
+
+type FifoLog struct {
```

Comment:
I think that variable names that contains acronyms probably should be capitalized (FifoLog -> FIFOLog)